### ISSUE TITLE
[Feature] npx playwright test should accept non-regex string

### ISSUE THREAD

**Issue by opack** [microsoft/playwright#28551]
Hi!

I use Playwright with SvelteKit. The routing system allows you to create directories whose names are enclosed in brackets. This makes it possible to organize files without having to include this part in the URL.
The problem is that Playwright doesn't seem to appreciate the existence of directories containing parentheses.

### System info
- Playwright Version: v1.39
- Operating System: Windows 10
- Browser: Chromium

### Source code
Place this simple test in a folder containing parentheses:
```typescript
// src/lib/(dbg)/dbg.spec.ts
import { test, expect } from '@playwright/test'

test('google test', async ({ page }) => {
	await page.goto('https://www.google.com/')
	await expect(page).toHaveURL('https://www.google.com/')
})
```
Here is the config file you can use to reproduce the issue:
```typescript
// playwright.config.ts
import type { PlaywrightTestConfig } from '@playwright/test'

const config: PlaywrightTestConfig = {
	testDir: './'
}

export default config
```

**Steps**/**Expected**/**Actual**
When I only mention the file name without the path containing the parentheses, everything is fine:
```shell
micro@BUREAU MINGW64 ~/Projects/bug
$ npx playwright test -c playwright.config.ts dbg.spec.ts

Running 1 test using 1 worker

ok 1 src\lib\(dbg)\dbg.spec.ts:3:1 › google test (1.3s)

1 passed (3.1s)
```
When I mention the file name with the path containing the parentheses, Playwright finds no tests:
```shell
micro@BUREAU MINGW64 ~/Projects/bug
$ npx playwright test -c playwright.config.ts "src/lib/(dbg)/dbg.spec.ts"
Error: No tests found
```

When I mention the file name and escape the parentheses in the path, everything is fine:
```shell
micro@BUREAU MINGW64 ~/Projects/bug
$ npx playwright test -c playwright.config.ts "src/lib/\(dbg\)/dbg.spec.ts"

Running 1 test using 1 worker

ok 1 src\lib\(dbg)\dbg.spec.ts:3:1 › google test (1.3s)

1 passed (3.0s)
```
I think it is related to Playwright using the parameter as a regular expression, in some way, because if I only escape one parenthesis (first or lasst) I have the following errors:
```shell
micro@BUREAU MINGW64 ~/Projects/bug
$ npx playwright test -c playwright.config.ts "src/lib/\(dbg)/dbg.spec.ts"
SyntaxError: Invalid regular expression: /src/lib/\(dbg)/dbg.spec.ts/: Unmatched ')'

micro@BUREAU MINGW64 ~/Projects/bug
$ npx playwright test -c playwright.config.ts "src/lib/(dbg\)/dbg.spec.ts"
SyntaxError: Invalid regular expression: /src/lib/(dbg\)/dbg.spec.ts/: Unterminated group
```

**Comment by dgozman** [comment]
> I think it is related to Playwright using the parameter as a regular expression, in some way

You are correct, Playwright treats command line arguments as regular expressions matching against the test file path. This is by design.

Let me see if we can improve the error message and documentation.

**Comment by opack** [comment]
Is there a way to have a command-line option to disable this behavior? Actually, I investigated this problem because when Playwright tests are executed by WebStorm, the command-line is generated. Thus, the only solution I see to work around my problem is telling WebStorm that it needs to append some option when it creates the command.

**Comment by mxschmitt** [comment]
> I investigated this problem because when Playwright tests are executed by WebStorm, the command-line is generated

Are you using [their Playwright integration](https://www.jetbrains.com/help/idea/playwright.html)? Or just copying the path and then running manually `npx playwright test`?

**Comment by opack** [comment]
Yes, I'm using the Test Automation plugin. But the problem also exists if I execute manually. That being said, both situations could be worked around with a command-line option ;-)

**Comment by mxschmitt** [comment]
Let me rephrase it into a feature request!

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
